webapp/player-project/src/main.ts

```diff
@@ -23,6 +26,18 @@ async function playSessionShadowing(gatewayAccessApi) {
   try {
     const recordingInfo = await gatewayAccessApi.fetchRecordingInfo();
     const fileType = getFileType(recordingInfo);
+    BeforeWebsocketClose((closeEvent) => {
+      if (closeEvent.code !== 1000) {
+        // faild, try to play static recording
+        cleanUpStreamers();
+        playStaticRecording(gatewayAccessApi);
+        showNotification('Session may have ended,playing recording instead', 'info');
```

Comment:
suggestion: Since this shown to the user, let’s observe strict rules about ponctuations (a comment must be followed by a space). That said, I suggest the following wording, what do you think?
 
```suggestion
        showNotification('Session already ended. The recording will be played from the start.', 'info');
```

We’ll have to think about internalization too if we’re really going to have messages displayed like this.

Reply:
True, I was thinking about emit the message back to the consumer, but it will be hard for RDM to catch it.

Reply:
I sent a Slack message in our Slack channel for further discussion about this!